src/main/java/com/zufar/icedlatte/review/api/ReviewDeleter.java

```diff
@@ -0,0 +1,27 @@
+package com.zufar.icedlatte.review.api;
+
+import com.zufar.icedlatte.review.repository.ReviewRepository;
+import com.zufar.icedlatte.security.api.SecurityPrincipalProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ReviewDeleter {
+
+    private final ReviewRepository reviewRepository;
+    private final SecurityPrincipalProvider securityPrincipalProvider;
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED)
+    public void delete(final UUID reviewId) {
+        securityPrincipalProvider.getUserId();
+        reviewRepository.deleteById(reviewId);
```

Comment:
you should compare the user ID of the commenter and the one who's trying to delete it

Reply:
Fixed